src/java.security.jgss/share/classes/sun/security/jgss/GSSUtil.java

```diff
@@ -320,56 +316,41 @@ public static boolean useMSInterop() {
-                        result = new Vector<>();
-                        Iterator<GSSCredentialImpl> iterator =
-                            currSubj.getPrivateCredentials
-                            (GSSCredentialImpl.class).iterator();
-                        while (iterator.hasNext()) {
-                            GSSCredentialImpl cred = iterator.next();
-                            if (DEBUG != null) {
-                                debug("...Found cred" + cred);
-                            }
-                            try {
-                                GSSCredentialSpi ce =
-                                    cred.getElement(mech, initiate);
-                                if (DEBUG != null) {
-                                    debug("......Found element: " + ce);
-                                }
-                                if (ce.getClass().equals(credCls) &&
-                                    (name == null ||
-                                     name.equals((Object) ce.getName()))) {
-                                    result.add(credCls.cast(ce));
-                                } else {
-                                    if (DEBUG != null) {
-                                        debug("......Discard element");
-                                    }
-                                }
-                            } catch (GSSException ge) {
-                                if (DEBUG != null) {
-                                    debug("...Discard cred (" + ge + ")");
-                                }
-                            }
+        Vector<T> creds = null;
+        Subject currSubj = Subject.current();
+        if (currSubj != null) {
+            creds = new Vector<>();
+            Iterator<GSSCredentialImpl> iterator =
+                currSubj.getPrivateCredentials
+                (GSSCredentialImpl.class).iterator();
+            while (iterator.hasNext()) {
+                GSSCredentialImpl cred = iterator.next();
+                if (DEBUG != null) {
+                    debug("...Found cred" + cred);
+                }
+                try {
+                    GSSCredentialSpi ce = cred.getElement(mech, initiate);
+                    if (DEBUG != null) {
+                        debug("......Found element: " + ce);
+                    }
+                    if (ce.getClass().equals(credCls) &&
+                        (name == null ||
+                         name.equals((Object) ce.getName()))) {
+                        creds.add(credCls.cast(ce));
+                    } else {
+                        if (DEBUG != null) {
+                            debug("......Discard element");
                         }
-                    } else if (DEBUG != null) {
-                        debug("No Subject");
                     }
-                    return result;
-                });
-            return creds;
-        } catch (PrivilegedActionException pae) {
```

Comment:
The `PrivilegedExceptionAction` never throws a checked exception, so it was not necessary to retain the code inside the catch block.